### PR TITLE
fix: connect exception to transaction

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs [hunk 1]
```diff
@@ -146,9 +152,20 @@ public async Task InvokeAsync(HttpContext context)
                         transaction.Name = transactionName;
                     }
 
-                    transaction.Finish(
-                        SpanStatusConverter.FromHttpStatusCode(context.Response.StatusCode)
-                    );
+                    var status = SpanStatusConverter.FromHttpStatusCode(context.Response.StatusCode);
+                    if (exception == null)
```
Comment (Tyrrrz) [inline src/Sentry.AspNetCore/SentryTracingMiddleware.cs:156]:
```suggestion
                    if (exception is null)
```

I thought we switched to this style now?

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs [hunk 2]
```diff
@@ -146,9 +152,20 @@ public async Task InvokeAsync(HttpContext context)
...
+                    {
+                        transaction.Finish(status);
+                    }
+                    else
+                    {
+                        transaction.Finish(exception, status);
+                    }
+                }
+
+                if (exception != null)
```
Comment (Tyrrrz) [inline src/Sentry.AspNetCore/SentryTracingMiddleware.cs:166]:
```suggestion
                if (exception is not null)
```

**Review by Tyrrrz — APPROVED** [review]

File: src/Sentry.AspNetCore/SentryTracingMiddleware.cs (hunk 1 above)
Comment (bruno-garcia) [reply src/Sentry.AspNetCore/SentryTracingMiddleware.cs:156]: yeah I missed it

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- connect exception to transaction ([#1043](https://github.com/getsentry/sentry-dotnet/pull/1043))
```
Comment (bruno-garcia) [inline CHANGELOG.md:5]:
```suggestion

### Fixes

- Connect middleware exceptions to transactions ([#1043](https://github.com/getsentry/sentry-dotnet/pull/1043))
```

**Pull request merged**